shared/utils/TasksProvider/TasksProvider.tsx

```diff
@@ -175,6 +176,18 @@ function TasksProvider({ children }: IProps) {
     updatedTask && setUndoStack([...storedTasksInUndo, updatedTask]);
   };
 
+  const saveChanges = async () => {
+    setUndoStack([[]]);
+    setRedoStack([[]]);
```

Comment:
Can place these inside the try catch so that error case does not reset the stack.